e2e/interchaintestv8/e2esuite/utils.go

```diff
@@ -409,3 +426,47 @@ func (s *TestSuite) BroadcastSdkTxBody(ctx context.Context, chain *cosmos.Cosmos
 
 	return s.BroadcastMessages(ctx, chain, user, gas, msgs...)
 }
+
+// GetRelayUpdateSlot extracts the latest update slot from the relay body's update messages.
+func (s *TestSuite) GetRelayUpdateSlot(chain *cosmos.CosmosChain, relayBody []byte) uint64 {
+	var txBody txtypes.TxBody
+	err := proto.Unmarshal(relayBody, &txBody)
+	s.Require().NoError(err)
+
+	var updateClientMsgsAny []*codectypes.Any
+	for _, msg := range txBody.Messages {
+		if msg.TypeUrl == "/ibc.core.client.v1.MsgUpdateClient" {
+			updateClientMsgsAny = append(updateClientMsgsAny, msg)
+		}
+	}
+	s.Require().NotEmpty(updateClientMsgsAny)
+
+	var headers []ethereumtypes.Header
+	for _, updateClientMsgAny := range updateClientMsgsAny {
+		var updateClientMsgSdkMsg sdk.Msg
+		err = chain.Config().EncodingConfig.InterfaceRegistry.UnpackAny(updateClientMsgAny, &updateClientMsgSdkMsg)
+		s.Require().NoError(err)
+
+		updateClientMsg, ok := updateClientMsgSdkMsg.(*clienttypes.MsgUpdateClient)
+		s.Require().True(ok)
```

Comment:
```suggestion
		var updateClientMsg clienttypes.MsgUpdateClient
		err = proto.Unmarshal(updateClientMsgAny.Value, &updateClientMsg)
		s.Require().NoError(err)
```
How does this feel worse? You are no longer using the chain after this. I can make this refactor if you'd like